src/demo-app/datepicker/datepicker-demo.html

```diff
@@ -133,3 +133,38 @@ <h2>Datepicker with value property binding</h2>
                     [startView]="yearView ? 'year' : 'month'"></mat-datepicker>
   </mat-form-field>
 </p>
+
+<h2>Datepicker emulating year picker</h2>
+<p>
+  <input matInput [matDatepicker]="datePicker5" class="datepicker-input-invisible">
```

Comment:
This hidden input stuff feels kind of hacky. What if we just use the `MomentDateAdapter` and provide a custom `MAT_DATE_FORMATS`

Our custom formats could just be a copy of [the defaults](https://github.com/angular/material2/blob/master/src/material-moment-adapter/adapter/moment-date-formats.ts) but with the parse and display formats for `dateInput` changed to `MM/YYYY` or `YYYY` respectively.

As for how to fit this into the demo, you could make some directives `demo-moment-month-year`, `demo-moment-year` that provide the appropriate things for their children

Reply:
I'll work on it.